examples/reshare/src/orchestrator/actor.rs

```diff
@@ -214,6 +216,16 @@ where
                         continue;
                     }
 
+                    if let Entry::Occupied(entry) = self.finalization_requests.entry(our_epoch) {
+                        if entry.get().1.elapsed() > FINALIZATION_REQUEST_TIMEOUT {
```

Comment:
I suppose the old approach led to us spamming a bunch of requests at the same time 🤔 . Feels a bit janky to have this request manager floating between the lines here...seems nice to have a utility where we register who reports they have an epoch and we just shuffle between?